flexxamples/howtos/openlayers.py

```diff
@@ -0,0 +1,177 @@
+    def remove_layers(self):
+        self.map.removeLayer(self.vectorLayer)
+        self.vectorLayer.getSource().clear()
+        self.vectorLayer.getSource().refresh()
+
+    @flx.action
+    def add_drawing_interaction(self):
+        self.map.addLayer(self.drawVectorLayer)
+        self.map.addInteraction(self.drawPoint)
+
+    @flx.action
+    def remove_drawing_interaction(self):
+        self.map.removeInteraction(self.drawPoint)
+        self.map.removeLayer(self.drawVectorLayer)
+        self.drawVectorLayer.getSource().clear()
+
+    @flx.action
+    def map_init(self):
+        global ol
+        self.olview = ol.View({
+            "zoom": 8,
+            "center": [-80.901813, 22.968599],
+            "projection": "EPSG:4326",
+            "minZoom": 3,
+            "maxZoom": 100
+        })
+
+        self.baseLayer = ol.layer.Tile({
+            "source": ol.source.OSM(),
+        })
+
+        self.vectorLayer = ol.layer.Vector({
+            "source": ol.source.Vector({
+                "format": ol.format.GeoJSON()
+            }),
+            "name": "Vector",
+            "style": ol.style.Style({
+                "image": ol.style.Circle({
+                    "radius": 7,
+                    "fill": ol.style.Fill({
+                        "color": 'rgba(255, 0, 0, 0.5)'
+                    })
+                })
+            }),
+        })
+
+        self.drawVectorLayer = ol.layer.Vector({
+            "source": ol.source.Vector({
+                "format": ol.format.GeoJSON()
+            }),
+            "name": "Draw Vector",
+            "style": ol.style.Style({
+                "image": ol.style.Circle({
+                    "radius": 7,
+                    "fill": ol.style.Fill({
+                        "color": 'rgba(0, 255, 0, 0.5)'
+                    })
+                })
+            }),
+        })
+
+        self.drawPoint = ol.interaction.Draw({
+            "type": 'Point',
+            "source": self.drawVectorLayer.getSource()
+        })
+
+        self.map_config = {
+            "target": self.mapnode,
+            'view': self.olview,
+            "controls": [ol.control.Zoom(), ol.control.MousePosition()],
+            "layers": []
+        }
+        self.map = ol.Map(self.map_config)
+        self.map.on('click', self.pointer_event)
+
+    @flx.emitter
+    def pointer_event(self, event):
+        return {"event": event}
+
+    @flx.action
+    def add_vector_layer(self):
+        format = self.vectorLayer.getSource().getFormat()
+        features = format.readFeatures(geojson)
+        self.vectorLayer.getSource().clear()
+        self.vectorLayer.getSource().addFeatures(features)
+        self.map.addLayer(self.vectorLayer)
+
+    @flx.action
+    def add_osm_layers(self):
+        self.map.addLayer(self.baseLayer)
+
+    def _create_dom(self):
+        """ This function gets automatically called when needed; Flexx is aware
+        of what properties are used here.
+        """
```

Comment:
This comment if false. The `_render_dom` is automatically called when any of the properties used within change. The `_create_dom` method is only called once (during initialization).

Reply:
ups, copy paste error. my bad